spotguides: SIGSEGV in (*SpotguideManager).scrapeSpotguides

**Describe the bug**
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2aa5dae]

goroutine 16 [running]:
github.com/banzaicloud/pipeline/spotguide.(*SpotguideManager).scrapeSpotguides(0xc00113d230, 0xc000fb2750, 0xc0012e4240, 0x1, 0x101)
	/Users/nandi/go/src/github.com/banzaicloud/pipeline/spotguide/spotguide.go:231 +0x1c7e
github.com/banzaicloud/pipeline/spotguide.(*SpotguideManager).ScrapeSharedSpotguides(0xc00113d230, 0xc0005fffa8, 0x1)
	/Users/nandi/go/src/github.com/banzaicloud/pipeline/spotguide/spotguide.go:198 +0x5a
main.main.func2(0xc00113d230, 0xc001186320)
	/Users/nandi/go/src/github.com/banzaicloud/pipeline/cmd/pipeline/main.go:305 +0x105
created by main.main
	/Users/nandi/go/src/github.com/banzaicloud/pipeline/cmd/pipeline/main.go:299 +0x1fb5
```

**Steps to reproduce the issue:**
Run pipeline for a certain number of minutes.

**Expected behavior**
No SIGSEGV.

--- a/spotguide/spotguide.go
+++ b/spotguide/spotguide.go
@@ -228,7 +228,7 @@ func (s *SpotguideManager) scrapeSpotguides(org *auth.Organization, githubClient
 
 		if err != nil {
 			// Empty organization, no repositories
-			if resp.StatusCode == http.StatusUnprocessableEntity {
+			if resp != nil && resp.StatusCode == http.StatusUnprocessableEntity {
 				return nil
 			}
 

